Neos.Flow/Classes/ResourceManagement/Target/FileSystemTarget.php

```diff
@@ -290,11 +314,19 @@ protected function publishFile($sourceStream, $relativeTargetPathAndFilename)
         }
 
         $targetPathAndFilename = $this->path . $relativeTargetPathAndFilename;
+        $streamMetaData = stream_get_meta_data($sourceStream);
+        $sourcePathAndFilename = $streamMetaData['uri'];
 
         if (@fstat($sourceStream) === false) {
```

Comment:
As we do a fstat here we might as well check the srouce stream path after this...

Reply:
I don't get it. If the `$sourceStream` is not available it stops here because of the exception

Reply:
Yes but until that point you don't need the two added lines above, only after. so I would move them after to avoid the extra work in case the fstat check fails. But I don't mind, it's a minor thing.